### PR TITLE
Fix block preview with sections

### PR DESCRIPTION
**Pull request by Prokyonn**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| License | MIT

#### What's in this PR?

Currently preview of collapsed blocks with sections doesn't work properly.
This PR removes the sections from the `blockSchemaTypeForm` for the block preview rendering and fixes this bug.

Additionally a check was added to skip potential preview fields without a value. So that the first fields with a value are shown in the preview.

#### Why?

Because the block preview should also work as properly with sections.

### CODE REVIEW

File: src/Sulu/Bundle/AdminBundle/Resources/js/containers/FieldBlocks/FieldBlocks.js [hunk 1]
```diff
@@ -167,12 +167,14 @@ class FieldBlocks extends React.Component<FieldTypeProps<Array<BlockEntry>>> {
...
         const blockSchemaType = this.getBlockSchemaType(type);
-        const blockSchemaTypeForm = blockSchemaType.form;
+        const blockSchemaTypeForm = this.removeSections(blockSchemaType.form);
 
         const previewPropertyNames = Object.keys(blockSchemaTypeForm)
             .filter((schemaKey) => {
                 const schemaEntryTags = blockSchemaTypeForm[schemaKey].tags;
-                return schemaEntryTags && schemaEntryTags.some((tag) => tag.name === BLOCK_PREVIEW_TAG);
+                return schemaEntryTags &&
+                    value[schemaKey] &&
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/containers/FieldBlocks/FieldBlocks.js:176]: Why is that necessary now?

File: src/Sulu/Bundle/AdminBundle/Resources/js/containers/FieldBlocks/FieldBlocks.js [hunk 2]
```diff
@@ -201,7 +203,7 @@ class FieldBlocks extends React.Component<FieldTypeProps<Array<BlockEntry>>> {
         if (previewPropertyNames.length === 0) {
             for (const fieldTypeKey of blockPreviewTransformerRegistry.blockPreviewTransformerKeysByPriority) {
                 for (const propertyName of Object.keys(blockSchemaTypeForm)) {
-                    if (blockSchemaTypeForm[propertyName].type === fieldTypeKey) {
+                    if (blockSchemaTypeForm[propertyName].type === fieldTypeKey && value[propertyName]) {
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/containers/FieldBlocks/FieldBlocks.js:206]: Why is that necessary?

File: src/Sulu/Bundle/AdminBundle/Resources/js/containers/FieldBlocks/FieldBlocks.js [hunk 3]
```diff
@@ -231,6 +233,20 @@ class FieldBlocks extends React.Component<FieldTypeProps<Array<BlockEntry>>> {
         );
     };
 
+    removeSections(blockSchemaTypeForm: Object) {
+        let filteredForm = {};
+        Object.keys(blockSchemaTypeForm).map((key) => {
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/containers/FieldBlocks/FieldBlocks.js:238]: The `map` call does not really make sense here, since it returns a new array, which is not used. We could use `forEach` instead, or use `reduce`, which would avoid the `let` variable at all:

```javascript
return Object.keys(blockSchemaTypeForm).reduce((blockSchema, key) => {
    if (blockSchemaTypeForm[key]['type'] === 'section') {
        blockSchema = {...blockSchema, ...this.removeSections(blockSchemaTypeForm[key]['items']};
    } else {
        filteredForm[key] = blockSchemaTypeForm[key];
    }

    return blockSchema;
}, {});
```

(Untested, might need small adjustements)

**Review by danrot — CHANGES_REQUESTED** [review]

File: src/Sulu/Bundle/AdminBundle/Resources/js/containers/FieldBlocks/FieldBlocks.js (hunk 1 above)
Comment (Prokyonn) [reply src/Sulu/Bundle/AdminBundle/Resources/js/containers/FieldBlocks/FieldBlocks.js:176]: It is not necessary for the block preview.

Currently the first fields are rendered no matter if these fields have values in it or not, with this change the first fields with a value are rendered.

**Review by danrot — APPROVED** [review]
